lib/spack/llnl/util/cpu/detect.py

```diff
@@ -75,30 +76,45 @@ def proc_cpuinfo():
     return info
 
 
-def check_output(args):
-    output = subprocess.Popen(args, stdout=subprocess.PIPE).communicate()[0]
+def check_output(args, env):
+    output = subprocess.Popen(
+        args, stdout=subprocess.PIPE, env=env
+    ).communicate()[0]
     return six.text_type(output.decode('utf-8'))
 
 
 @info_dict(operating_system='Darwin')
 def sysctl():
     """Returns a raw info dictionary parsing the output of sysctl."""
+    # Make sure that /sbin and /usr/sbin are in PATH as sysctl is
+    # usually found there
+    child_environment = dict(os.environ.items())
+    search_paths = child_environment.get('PATH', '').split(os.pathsep)
+    for additional_path in ('/sbin', '/usr/sbin'):
+        if additional_path not in search_paths:
+            search_paths.append(additional_path)
+    child_environment['PATH'] = os.pathsep.join(search_paths)
```

Comment:
Suggested refactor: call this function `sysctl_info_dict()` or something more descriptive, then make an internal wrapper around `check_output()` called `sysctl()`:

```python
@info_dict(operating_system='Darwin')
def sysctl_info_dict():
    child_environment = dict(os.environ.items())
    search_paths = child_environment.get('PATH', '').split(os.pathsep)
    for additional_path in ('/sbin', '/usr/sbin'):
        if additional_path not in search_paths:
            search_paths.append(additional_path)
    child_environment['PATH'] = os.pathsep.join(search_paths)

    def sysctl(*args):
        return check_output(['sysctl'] + args, env=child_environment)

    ...
```

Reply:
Seems indeed a lot better. I'll probably try to see if turning `info` into a dict literal improves readability.

Reply:
Done